quantities/quantity.py

```diff
@@ -283,11 +290,12 @@ def __array_prepare__(self, obj, context=None):
                 )
         return res
 
-    def __array_wrap__(self, obj, context=None):
+    def __array_wrap__(self, obj, context=None, return_scalar=False):
         if not isinstance(obj, Quantity):
             # backwards compatibility with numpy-1.3
-            obj = self.__array_prepare__(obj, context)
-        return obj
+            return self.__array_prepare__(obj, context)
+        else:
+            return super().__array_wrap__(obj, context, return_scalar)
```

Comment:
I'm confused by this. The previous array wrap would only do something if the object was not a Quantity if the object was already a quantity it would just return the object without any changes. Here you force the __array_wrap__ even in the case of the object being a Quantity. It seems like for the case of some failing tests the objects are losing their dimensionality and this seems like it could be a place where a Quantity should be returned, no?

I'm not super familiar with this deep level of array wrapping so just thought maybe a discussion of this could help point us to the actual test failure issue.

Reply:
@bjodah just wanted to bump my message. Does this seem like it could be the problem with the tests failing?